app/ui-react/syndesis/src/modules/integrations/components/editor/AddStepPage.tsx

```diff
@@ -160,6 +160,14 @@ export const AddStepPage: React.FunctionComponent<
                     getFirstPosition(state.integration, params.flowId) ||
                   position === getLastPosition(state.integration, params.flowId)
                 ) {
+                  setIsDeleting(true);
+                  const newInt = await removeStep(
+                    state.integration,
+                    params.flowId,
+                    position!
+                  );
+                  state.integration = newInt;
```

Comment:
Sorry, the state object shouldn’t be mutated, ever.

Reply:
Would this work:
```
                  history.push(
                    getDeleteEdgeStepHref(
                      position!,
                      params, {
                      ...state,
                      integration: newInt,
                    })
                  );
```
Something like that is done when a non-endpoint is deleted. Is that what you had in mind @riccardo-forina?

Reply:
Well that's not mutating the `state` object, it's creating a new object that holds also the values from the `state`. But the thing is, you're trying to modify the `integration` object that's describing the _current_ value, but you can't be sure the user will conclude the delete process.

In the following pages, links to go back and abort the action will use the integration object in the route state to say "this is the integration, unchanged"; but with this change, this assertion will not be true anymore.